server/server.go

```diff
@@ -421,6 +420,7 @@ func (s *Server) Start() error {
 	s.node.startWriteSummaries(s.ctx.MetricsSampleInterval)
 
 	s.sqlExecutor.SetNodeID(s.node.Descriptor.NodeID)
```

Comment:
I forgot how this works; when does is this `s.node.NodeID` assigned? This seems like a late place to init these fields on the executor and distsqlServer.